app/src/test/scala/org/alephium/explorer/persistence/queries/EventQueriesSpec.scala

```diff
@@ -0,0 +1,165 @@
+      forAll(Gen.nonEmptyListOf(eventEntityGen)) { events =>
+        insert(events)
+
+        events.map { event =>
+          val result =
+            run(EventQueries.getEventsByContractAddressQuery(event.contractAddress)).futureValue
+          result.size is 1
+          result.head.toApi is event.toApi
+        }
+      }
+    }
+
+    "get all events with same contractAddress" in new Fixture {
+      forAll(Gen.nonEmptyListOf(eventEntityGen)) { events =>
+        val contractAddress = addressGen.sample.get
+        val uniqueContractAddressEvents = events.map { event =>
+          event.copy(contractAddress = contractAddress)
+        }
+
+        insert(uniqueContractAddressEvents)
+
+        val result =
+          run(EventQueries.getEventsByContractAddressQuery(contractAddress)).futureValue
+
+        result.size is uniqueContractAddressEvents.size
+        result.zip(uniqueContractAddressEvents.sortBy(_.timestamp).reverse).map {
+          case (res, event) =>
+            res.toApi is event.toApi
+        }
+      }
+    }
+
+    "get event by contract address and input address" in new Fixture {
+      forAll(Gen.nonEmptyListOf(eventEntityGen)) { events =>
+        insert(events)
+
+        events.map { event =>
+          event.inputAddress match {
+            case Some(inputAddress) =>
+              val result =
+                run(
+                  EventQueries.getEventsByContractAndInputAddressQuery(event.contractAddress,
+                                                                       inputAddress)).futureValue
+
+              result.size is 1
+              result.head.toApi is event.toApi
+            case None =>
+              run(
+                EventQueries.getEventsByContractAndInputAddressQuery(
+                  event.contractAddress,
+                  addressGen.sample.get)).futureValue is ArraySeq.empty
+
+          }
+        }
+      }
+    }
+
+    "get all events with same contractAddress and input address" in new Fixture {
+      forAll(Gen.nonEmptyListOf(eventEntityGen)) { events =>
+        val contractAddress = addressGen.sample.get
+        val inputAddress    = addressGen.sample.get
+        val uniqueContractAddressEvents = events.map { event =>
+          event.copy(contractAddress = contractAddress, inputAddress = Some(inputAddress))
+        }
+
+        insert(uniqueContractAddressEvents)
+
+        val result =
+          run(EventQueries.getEventsByContractAndInputAddressQuery(contractAddress, inputAddress)).futureValue
+
+        result.size is uniqueContractAddressEvents.size
+        result.zip(uniqueContractAddressEvents.sortBy(_.timestamp).reverse).map {
+          case (res, event) =>
+            res.toApi is event.toApi
+        }
+      }
+    }
+  }
+
+  trait Fixture {
+    def insert(events: ArraySeq[EventEntity]) = {
+      run(EventSchema.table.delete).futureValue
+      run(EventSchema.table ++= events).futureValue
+    }
+  }
```

Comment:
Following issue #368 should these basic queries be moved to their dedicated `EventQueries` object?

This `Fixture` not being stateful means there's no need for the `Fixture` right? We could pull out `insert` to just a plain function sitting somewhere in the test?

Reply:
oh indeed! yes let's try to follow #368 

you right with the Fixture, I have a bit of a reflex to use them, but it's not needed here.

Reply:
oh still need to extract it out the `Fixture`